.github/workflows/haskell.yml

```diff
@@ -5,38 +5,25 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v1
-    - uses: cachix/install-nix-action@v7
-    - uses: cachix/cachix-action@v4
+    - uses: mstksg/setup-stack@v1
+    - uses: actions/cache@v1
       with:
-        name: 47deg
-        signingKey: '${{ secrets.CACHIX_SIGNING_KEY }}'
-
-  test_stack_templates:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: mstksg/setup-stack@v1
-
-      - uses: actions/checkout@v1
-
-      - uses: actions/cache@v1
-        with:
-          path: ~/.stack
-          key: dot-stack-${{ hashFiles('templates/*.hsfiles') }}
-          restore-keys: |
-            dot-stack-
-
-      - name: Generate Avro RPC server from Stack template
-        run: stack new muhaskellavro https://raw.githubusercontent.com/higherkindness/mu-haskell/master/templates/grpc-server-avro.hsfiles -p "author-email:haskell.curry@47deg.com" -p "author-name:Haskell Curry"
-        working-directory: /tmp
-
-      - name: Build the generated Avro RPC server
-        run: stack build
-        working-directory: /tmp/muhaskellavro
-
-      - name: Generate Protobuf RPC server from Stack template
-        run: stack new muhaskellprotobuf https://raw.githubusercontent.com/higherkindness/mu-haskell/master/templates/grpc-server-protobuf.hsfiles -p "author-email:haskell.curry@47deg.com" -p "author-name:Haskell Curry"
-        working-directory: /tmp
-
-      - name: Build the generated Protobuf RPC server
-        run: stack build
-        working-directory: /tmp/muhaskellprotobuf
+        path: ~/.stack
+        key: dot-stack-${{ hashFiles('stack.yaml') }}
+        restore-keys: |
+          dot-stack-
+    - name: Compile with Stack
+      run: stack build
+    # Test Stack Templates
```

Comment:
You don't want to run this as a separate job? I prefer to split the workflow into fine-grained jobs so that a) they run in parallel, and b) it's easy to see what failed without scrolling through 1000s of lines of logs. But it's up to you - it's your project :)

Reply:
Makes sense, thanks! 😄